system/HTTP/Response.php

```diff
@@ -428,13 +420,18 @@ public function setJSON($body)
 	 * Returns the current body, converted to JSON is it isn't already.
 	 *
 	 * @return mixed|string
+	 *
+	 * @throws \InvalidArgumentException If the body property is not array.
 	 */
 	public function getJSON()
 	{
 		$body = $this->body;
 
 		if ($this->bodyFormat !== 'json')
 		{
+			/**
+			 * @var Format $config
+			 */
 			$config    = config(Format::class);
 			$formatter = $config->getFormatter('application/json');
 
```

Comment:
In this file, in the line below, `$formatter->format()` is typed to expect an array.

 https://github.com/codeigniter4/CodeIgniter4/blob/cd8eba1fa72c45c199b5f0d3e91e2fdac91295b6/system/Format/FormatterInterface.php#L49

Removing the type hint in the FormatterInterface allow more flexibility to work with objects, like pass an `\CodeIgniter\Entity` directly.

The conversion could be done inside the method, if necessary:

https://github.com/codeigniter4/CodeIgniter4/blob/cd8eba1fa72c45c199b5f0d3e91e2fdac91295b6/system/Format/XMLFormatter.php#L61

JSONFormatter does not needs.

Reply:
Interesting idea, and I'm ok with that. You could easily check if it's an `Entity` and call it's `toArray()` method to get the values.